evm/vm/base.py

```diff
@@ -674,10 +681,20 @@ def validate_block(self, block):
                 )
             )
 
+    def validate_seal(self, header: BlockHeader) -> None:
+        """
+        Validate the seal on the given header.
+        """
+        check_pow(
+            header.block_number, header.mining_hash,
+            header.mix_hash, header.nonce, header.difficulty)
+
     def validate_uncle(self, block, uncle):
         """
         Validate the given uncle in the context of the given block.
         """
+        self.validate_seal(block.header)
```

Comment:
I pushed an update to the test so we can see the stack trace that failed the block. It points to this line. Maybe there are some paths that call `validate_uncle` before this block's mixhash has been finalized?

Reply:
Tried to execute the `pytest` individually per your suggestion and I'm getting this: https://gist.github.com/glaksmono/8c273fe4bdd01d2396f95ae36df29794 - thoughts?

Reply:
```
...
  /Users/glaksmono/Documents/py-evm/venv/lib/python3.6/site-packages/pluggy/callers.py(180)_multicall()
-> res = hook_impl.function(*args)
  /Users/glaksmono/Documents/py-evm/venv/lib/python3.6/site-packages/_pytest/python.py(156)pytest_pyfunc_call()
-> testfunction(**testargs)
  /Users/glaksmono/Documents/py-evm/tests/json-fixtures/test_blockchain.py(103)test_blockchain_fixtures()
-> (block, mined_block, block_rlp) = apply_fixture_block_to_chain(block_fixture, chain)
  /Users/glaksmono/Documents/py-evm/evm/tools/fixture_tests.py(632)apply_fixture_block_to_chain()
-> mined_block = chain.import_block(block)
  /Users/glaksmono/Documents/py-evm/evm/chains/base.py(546)import_block()
-> imported_block = parent_chain.get_vm().import_block(block)
  /Users/glaksmono/Documents/py-evm/evm/vm/base.py(428)import_block()
-> return self.mine_block()
  /Users/glaksmono/Documents/py-evm/evm/vm/base.py(434)mine_block()
-> block = self.pack_block(self.block, *args, **kwargs)
  /Users/glaksmono/Documents/py-evm/evm/vm/base.py(528)pack_block()
-> self.validate_block(packed_block)
  /Users/glaksmono/Documents/py-evm/evm/vm/base.py(661)validate_block()
-> self.validate_uncle(block, uncle)
  /Users/glaksmono/Documents/py-evm/evm/vm/base.py(695)validate_uncle()
-> self.validate_seal(block.header)
  /Users/glaksmono/Documents/py-evm/evm/vm/base.py(689)validate_seal()
-> header.mix_hash, header.nonce, header.difficulty)
> /Users/glaksmono/Documents/py-evm/evm/consensus/pow.py(65)check_pow()
-> encode_hex(mining_output[b'mix digest']), encode_hex(mix_hash)))
```
So the reason why `validate_uncle` is called prior to the `validate_seal` here is because I put `validate_seal` within `validate_uncle`

But that is the [first](https://github.com/glaksmono/py-evm/blob/021d244d86a6c695720425e3b7d83ff953fca947/evm/vm/base.py#L695) thing to be executed in `validate_uncle`

Prior to this is all "python related stuff"

Reply:
>  I put validate_seal(block.header) within validate_uncle

This change repeatedly checks the block header for each uncle. There should be a separate check anyway, in case there are 0 uncles. What's the benefit of putting it there?